nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -547,6 +506,173 @@ public void compareGenericTypeParameterNullabilityForCall(
+          // running
+          // NullAway
+          if (rhsType == null) {
+            throw new RuntimeException(
+                "Did not find supertype of " + rhsType + " matching " + lhsType);
+          }
+          List<Type> lhsTypeArguments = lhsType.getTypeArguments();
+          List<Type> rhsTypeArguments = rhsType.getTypeArguments();
+          // This is impossible, considering the fact that standard Java subtyping succeeds before
+          // running
+          // NullAway
+          if (lhsTypeArguments.size() != rhsTypeArguments.size()) {
+            throw new RuntimeException(
+                "Number of types arguments in " + rhsType + " does not match " + lhsType);
+          }
+          for (int i = 0; i < lhsTypeArguments.size(); i++) {
+            Type lhsTypeArgument = lhsTypeArguments.get(i);
+            Type rhsTypeArgument = rhsTypeArguments.get(i);
+            boolean isLHSNullableAnnotated = false;
+            List<Attribute.TypeCompound> lhsAnnotations = lhsTypeArgument.getAnnotationMirrors();
+            // To ensure that we are checking only jspecify nullable annotations
+            for (Attribute.TypeCompound annotation : lhsAnnotations) {
+              if (annotation.getAnnotationType().toString().equals(NULLABLE_NAME)) {
+                isLHSNullableAnnotated = true;
+                break;
+              }
+            }
+            boolean isRHSNullableAnnotated = false;
+            List<Attribute.TypeCompound> rhsAnnotations = rhsTypeArgument.getAnnotationMirrors();
+            // To ensure that we are checking only jspecify nullable annotations
+            for (Attribute.TypeCompound annotation : rhsAnnotations) {
+              if (annotation.getAnnotationType().toString().equals(NULLABLE_NAME)) {
+                isRHSNullableAnnotated = true;
+                break;
+              }
+            }
+            if (isLHSNullableAnnotated != isRHSNullableAnnotated) {
+              return false;
+            }
+            // nested generics
+            List<Type> genericArgs = lhsTypeArgument.accept(TYPE_ARG_VISITOR, null);
+            if (genericArgs.size() > 0) {
+              if (!lhsTypeArgument.accept(COMPARE_NULLABILITY_VISITOR, rhsTypeArgument)) {
+                return false;
+              }
+            }
+          }
+          return true;
+        }
+
+        @Override
+        public Boolean visitArrayType(Type.ArrayType lhsType, Type rhsType) {
+          Type.ArrayType arrRhsType = (Type.ArrayType) rhsType;
+          return lhsType
+              .getComponentType()
+              .accept(COMPARE_NULLABILITY_VISITOR, arrRhsType.getComponentType());
+        }
+
+        @Override
+        @Nullable
+        public Boolean visitType(Type t, Type type) {
+          return null;
+        }
+      };
+
+  /** Visitor For Handling Different Tree Types */
+  public class PreservedAnnotationTreeVisitor extends SimpleTreeVisitor<Type, Boolean> {
```

Comment:
Why is it `SimpleTreeVisitor<Type, Boolean>` and not `SimpleTreeVisitor<Type, Void>`?  What is the `Boolean` used for?

Reply:
It should be Void, I do not recall why it was Boolean but it certainly doesn't make sense now, Thanks for catching that.